### ISSUE TITLE
Downloads failing 

### ISSUE THREAD

**Issue by thaussma** [wouterdebie/putioarr#8]
### Describe the bug
Downloads from put.io are failing kind of randomly. Some files work right away and some have to be retried several times for success and others never finish successfully.
If a file stops/fails to download, putioarr just downloads the next file. If all files are have been tried once, failed files are not automatically retried. Restarting putioarr will restart the downloads and sometimes some of the files that had failed before are downloaded successfully. 

### Logs

> [2024-01-30T11:43:20Z ERROR] [5113: /downloads/AAAAA.mkv]: download failed: request or response body error: error reading a body from connection: stream error received: unspecific protocol error detected

**Comment by thaussma** [comment]
When using `curl` to download from put.io I observed that most downloads stalled and then just timed out. However `curl` did retry and resume the downloads and because of this I managed to download whole files after resuming a few times.

`putioarr` does not retry and resume downloads and therefore a lot of downloads were aborted/partial.  And `putioarr` need to be restarted in order to start downloading again.
